src/picongpu/include/plugins/output/GatherSlice.hpp

```diff
@@ -55,7 +62,9 @@ struct GatherSlice
      */
     bool init(bool isActive)
     {
-        /*free old communicator of init is called again*/
+        static int masterRankOffset = 0;
```

Comment:
Thread safe in the context of function local statics refers to their initialization.
If the static object is not a fundamental type but a class with a constructor it is not guaranteed that two threads calling the function containing the static object will result in exactly one constructor call. The constructor could be called multiple times and arbitrary evil things could happen.